src/components/Calculator.js

```diff
@@ -1,41 +1,55 @@
 import React from 'react';
+import calculate from '../logic/calculate';
 
 class Calculator extends React.Component {
-  calcHtml = (
-    <div className="calc-wrapper">
-      <input type="text" className="inp-display" placeholder={0} />
-      <div className="grid">
-        <div className="calc-btn"> AC </div>
-        <div className="calc-btn"> +/- </div>
-        <div className="calc-btn"> % </div>
-        <div className="calc-btn op"> + </div>
-        <div className="calc-btn"> 7 </div>
-        <div className="calc-btn"> 8 </div>
-        <div className="calc-btn"> 9 </div>
-        <div className="calc-btn op"> * </div>
-        <div className="calc-btn"> 4 </div>
-        <div className="calc-btn"> 5 </div>
-        <div className="calc-btn"> 6 </div>
-        <div className="calc-btn op"> - </div>
-        <div className="calc-btn"> 1 </div>
-        <div className="calc-btn"> 2 </div>
-        <div className="calc-btn"> 3 </div>
-        <div className="calc-btn op"> + </div>
-        <div className="calc-btn zero-btn"> 0 </div>
-        <div className="calc-btn"> . </div>
-        <div className="calc-btn op"> = </div>
-      </div>
-    </div>
-  );
-
   constructor(props) {
     super(props);
-    this.state = {};
+    this.state = { total: null, next: null, operation: null };
   }
 
+  eventHandler = (e) => {
+    const data = e.target.innerHTML;
+    const calcObj = calculate(this.state, data);
+    this.setState(calcObj);
+  };
+
   render() {
-    return this.calcHtml;
+    const { total, next, operation } = this.state;
+    return (
+      <div className="calc-wrapper">
+        <div className="inp-display">
+
+          {total}
+
+          {next}
+
+          {operation}
```

Comment:
- [ ] The concern of @Whoistolu is that the display is misleading and not making sense right now. Please consider refactoring these lines as suggested below. Let `{operation}` be between `{total}` and `{next}`. This will help address the concern that @Whoistolu had.

```JavaScript
          {total}

          {operation}

          {next}
```

Reply:
@Lameck1 I was going to put an inline comment on the `calculate.js` component and not the logic. This was an error due to the same file name. Thank you for pointing this out.